src/main/java/com/ita/if103java/ims/service/impl/WarehouseServiceImpl.java

```diff
@@ -180,11 +180,19 @@ public boolean softDelete(Long id, UserDetailsImpl user) {
 
     private void createEvent(UserDetailsImpl user, Warehouse warehouse, EventName eventName) {
         Event event = new Event();
+        int level;
+        Long parentId = warehouse.getParentID();
         String message = eventName.getLabel();
-        int level = warehouseDao.findLevelByParentID(warehouse.getParentID());
-        if (eventName == EventName.WAREHOUSE_CREATED) {
-            message += " Name : " + warehouse.getName() + "level : " + level;
+        if (parentId == null) {
+            level = 0;
+
+        } else {
+            level = warehouseDao.findLevelByParentID(parentId);
+            if (eventName == EventName.WAREHOUSE_CREATED) {
+                message += " Name : " + warehouse.getName() + "level : " + level;
+            }
         }
+        message += " Name : " + warehouse.getName() + "level : " + level;
```

Comment:
This code works wrongly. By my mind to work correctly, please do as following
```
int level = 0;
if (warehouse.getParentID() != null) {
  level = warehouseDao.findLevelByParentID(warehouse.getParentID());
}
message += " Name : " + warehouse.getName() + "level : " + level;
```

I do not understand, why you add such condition 
```
if (eventName == EventName.WAREHOUSE_CREATED) {
                message += " Name : " + warehouse.getName() + "level : " + level;
            }
```
As for me it is not needed